site/tests/integration/hubpage.test.ts

```diff
@@ -0,0 +1,83 @@
+import { expect, test } from "@playwright/test";
+
+import blocksData from "../../blocks-data.json";
+import type { ExpandedBlockMetadata } from "../../src/lib/blocks";
+
+test("Hub page should contain key elements", async ({ page }) => {
+  await page.goto("/hub");
+
+  await expect(
+    page.locator(
+      "h1 >> text=Interactive, data-driven blocks to use in your projects",
+    ),
+  ).toBeVisible();
+
+  await expect(
+    page.locator("text=All open-source and free to use"),
+  ).toBeVisible();
+
+  expect(
+    await page.locator('[data-testid="block-card"]').count(),
+  ).toBeGreaterThan(5);
+
+  await expect(page.locator("text=Quickstart guide")).toHaveAttribute(
+    "href",
+    "/docs/developing-blocks",
+  );
+
+  await expect(page.locator("text=Read the Quick Start Guide")).toHaveAttribute(
+    "href",
+    "/docs/developing-blocks",
+  );
+});
+
+const codeBlock = blocksData.find(
+  ({ name }) => name === "@hashintel/block-code",
+) as ExpandedBlockMetadata | null;
+
+test("Block Card should contain key elements", async ({ page }) => {
+  expect(
+    codeBlock,
+    "Code block should be prepared before this test",
+  ).toBeDefined();
+
+  if (!codeBlock) {
+    return;
+  }
```

Comment:
That’s just to narrow TS type, right? How about throwing an exception `if (!codeBlock)` instead? We might even be able to do this in the root of the file. Not having block metadata does not fail the test but rather fails the setup.

Reply:
I agree. I initially put it in the test because we don't need it for all tests in that file. will update